feat: Initial processing and uploading to nmkr api

KristijanBoshev opened the pull request:
This commit serves as an Initial code for uploading NFTs and metadata for each one of them through NMKR Api. Next step is scalling to 10,000 uploads effectively in one-go. Here is proof of the uploaded test NFTs and metadata. 
![proof1](https://github.com/user-attachments/assets/80160bbd-88c2-4906-8b59-6a826ca1c2e5)
![proof2](https://github.com/user-attachments/assets/72720eef-6a2a-4b47-93e3-68d327973051)

goran0706 reviewed (commented):
**General Improvements:**
- Remove `.DS_Store` from the repository - this is unnecessary clutter.

**Timeout Configuration:**
- Avoid setting multiple timeout configurations across different layers (serverless, serverless plugins, axios).
- Prefer default values and only override when absolutely necessary.
- Ensure single configuration to prevent inconsistencies during development.
- Ensure a clear understanding of the serverless config before applying custom values.
- Remove the axios timeout; its default of 0 (no timeout) should be sufficient.

**Environment Variables Validation:**
- Ensure all required environment variables are set - use `validateEnvVar` helper for consistency and maintainability.

**Multer Middleware:**
- Review the changes for the multer middleware - it seems the current logic may break single file uploads.
- Handle both single and array uploads.

**Metadata Handling:**
- Avoid overriding the metadata for each NFT with metadataOverride: JSON.stringify(cip25Metadata).
- Utilize the Cip25Metadata interface for type safety and adherence to the CIP25 standard.
- There is no explicit need to override project-wide metadata.
- Refer to the API documentation and ensure the correct payload is sent - specifically, see usage for `metadataPlaceholder` instead of `metadataOverride`
- "With this API you can upload a file to IPFS and add it to a project. You can upload the file as BASE64 Content or as URL Link or as IPFS Hash. If you submit Metadata, this Metadata will be used instead of the Metadatatemplate from the project. You can either submit Metadata or MetadataPlaceholder, but not both (because it makes no sense). The Metadata field is optional and if you dont use it, it will use the Template from your project. It is poosible to mix both versions in one project. You can have one nft with own metadata and other nfts with the template."

```js

"metadataPlaceholder": [
  {
    "name": "string",
    "value": "string"
  }
]
```

**Unnecessary Interfaces:**
- Remove PlaceholderCSV and FieldData as they no longer serve a purpose.
- Set asset_name to "unknown" to dynamically handle CSV headers.

**Directory Structure & Naming Conventions**
- Move aws_clients.ts to the core directory - this aligns better with its purpose.
- Remove it from the utils folder to prevent clutter and confusion.
- Ensure proper file naming conventions for Node.js: Use PascalCase for class-based modules or Use dot notation (e.g., file.client.ts) for service and utility files.

**NFT Preview and Upload Logic:**
- Review the NFT preview logic in the API docs; currently, the "View assets" shows IDs instead of images.
- Understand how different upload methods are handled:
- - IPFS (fileFromIPFS)
- - URL (e.g., S3 Bucket - fileFromsUrl)
- - Base64 (fileFromBase64) for fully on-chain uploads.
```js
"previewImageNft": {
  "mimetype": "string",
  "fileFromBase64": "string",
  "fileFromsUrl": "string",
  "fileFromIPFS": "string"
}
```
- When constructing the payload, use more descriptive naming instead of `params`.
- - Consider renaming to fileUploadPayload for clarity
- - Use singluar naming if it represent a single file UploadFiles
```js
const fileUploadPayload: UploadFiles = {
  tokenname: uuid,
  previewImageNft: {
    mimetype: mimetype,
    fileFromBase64: fileFromBase64,
  },
  metadataPlaceholder: cip25Metadata,
};
```

- Consider more scalable solution like SFTP/FTP for large file uploads instead of chunking it

KristijanBoshev commented:
Commit `9228ae1` references the following changes:

Added DS_Store in .gitignore
Remove all necesery timeouts
All .env variables and validated with `validateEnvVar`
Multer both for single and array of files
Removed PlaceholderCsv and FieldData interfaces
Set asset_name to "unknown" to dynamically handle CSV headers.
Move aws_clients.ts to the core directory - this aligns better with its purpose.
Remove it from the utils folder to prevent clutter and confusion.
Ensure proper file naming conventions for Node.js: Use PascalCase for class-based modules or Use dot notation (e.g., file.client.ts) for service and utility files.

All other comments regarding the NMKR API usage are not relevant since that implementation will be substituted with more scalable one (SFTP + SQS) which is currently under development

goran0706 commented:
- Ensure Git stops tracking the file and add it to .gitignore.
- Verify existing environment variables and configuration. Reuse the same bucket or define a new name if necessary. Check for bucketKey in the config; if there is no corresponding .env variable, avoid duplication. This one is too generic (BUCKET_NAME) 
- Remove bucketKey if it's not needed.
- Optimize file uploads by processing them in parallel and implementing request throttling. Alternatively, consider using FTP for better efficiency.
